scripts/Connector/Connector_Hirose/conn_hirose_df11_tht_top.py

```diff
@@ -0,0 +1,239 @@
+#!/usr/bin/env python3
+
+import sys
+import os
+#sys.path.append(os.path.join(sys.path[0],"..","..","kicad_mod")) # load kicad_mod path
+
+# export PYTHONPATH="${PYTHONPATH}<path to kicad-footprint-generator directory>"
+sys.path.append(os.path.join(sys.path[0], "..", "..", ".."))  # load parent path of KicadModTree
+from math import sqrt
+import argparse
+import yaml
+from helpers import *
+from KicadModTree import *
+
+sys.path.append(os.path.join(sys.path[0], "..", "..", "tools"))  # load parent path of tools
+from footprint_text_fields import addTextFields
+
+series = ""
+series_long = 'DF11 through hole'
+manufacturer = 'Hirose'
+orientation = 'V'
+number_of_rows = 2
+datasheet = 'https://www.hirose.com/product/document?clcode=&productname=&series=DF11&documenttype=Catalog&lang=en&documentid=D31688_en'
+
+#pins_per_row per row
+pins_per_row_range = range(2,17)
+
+#Molex part number
+#n = number of circuits per row
+part_code = "DF11-{n}DP-2DSA"
+
+pitch = 2
+drill = 0.8
+
+pad_to_pad_clearance = 0.8
+max_annular_ring = 0.4
+min_annular_ring = 0.15
+
+
+
+pad_size = [pitch - pad_to_pad_clearance, drill + 2*max_annular_ring]
+if pad_size[0] - drill < 2*min_annular_ring:
+    pad_size[0] = drill + 2*min_annular_ring
+if pad_size[0] - drill > 2*max_annular_ring:
+    pad_size[0] = drill + 2*max_annular_ring
+
+pad_shape=Pad.SHAPE_OVAL
+if pad_size[1] == pad_size[0]:
+    pad_shape=Pad.SHAPE_CIRCLE
+
+
+
+def generate_one_footprint(pins, configuration):
+    mpn = part_code.format(n=pins*2)
+    pad_silk_off = configuration['silk_line_width']/2 + configuration['silk_pad_clearance']
+    # handle arguments
+    orientation_str = configuration['orientation_options'][orientation]
+    footprint_name = configuration['fp_name_format_string'].format(man=manufacturer,
+        series=series,
+        mpn=mpn, num_rows=number_of_rows, pins_per_row=pins, mounting_pad = "",
+        pitch=pitch, orientation=orientation_str)
+
+    footprint_name = footprint_name.replace("__",'_')
+
+    kicad_mod = Footprint(footprint_name)
+    kicad_mod.setDescription("Molex {:s}, {:s}, {:d} Pins per row ({:s}), generated with kicad-footprint-generator".format(series_long, mpn, pins_per_row, datasheet))
+    kicad_mod.setTags(configuration['keyword_fp_string'].format(series=series,
+        orientation=orientation_str, man=manufacturer,
+        entry=configuration['entry_direction'][orientation]))
+
+    A = (pins - 1) * pitch
+    B = A + 4
+
+    # create pads
+    #createNumberedPadsTHT(kicad_mod, pincount, pitch, drill, {'x':x_dia, 'y':y_dia})
+
+    optional_pad_params = {}
+    if configuration['kicad4_compatible']:
+        optional_pad_params['tht_pad1_shape'] = Pad.SHAPE_RECT
+    else:
+        optional_pad_params['tht_pad1_shape'] = Pad.SHAPE_ROUNDRECT
+
+    kicad_mod.append(PadArray(start=[0, 0], pincount=pins, increment=2, x_spacing=pitch,
```

Comment:
You can do this in a single loop with zip(), like `for initial, initial_y in zip([1, 2], [0, pitch]):`. You can remove lines 83-86 and put this before line 88, and then change the argument `initial` to `initial` instead of `2`. Easier to maintain.